library/src/main/java/com/novoda/downloadmanager/PartialDownloadMigrationExtractor.java

```diff
@@ -0,0 +1,64 @@
+package com.novoda.downloadmanager;
+
+import android.database.Cursor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class PartialDownloadMigrationExtractor {
+
+    private static final String BATCHES_QUERY = "SELECT batches._id, batches.batch_title FROM batches INNER JOIN DownloadsByBatch ON " +
```

Comment:
I think we can make this query a bit more intelligent. Would rather merge and make improvements in a follow-up PR today.

Reply:
do we not need `SELECT DISTINCT` here?

Reply:
No, you can do the same with a `GROUP_BY` 😄